saleor/tests/e2e/orders/test_expired_order_is_deleted_after_specified_time.py

```diff
@@ -0,0 +1,124 @@
+        permission_manage_products,
+        permission_manage_channels,
+        permission_manage_shipping,
+        permission_manage_product_types_and_attributes,
+        permission_manage_orders,
+    ]
+    assign_permissions(e2e_staff_api_client, permissions)
+    app_permissions = [
+        permission_manage_payments,
+        permission_handle_checkouts,
+    ]
+    assign_permissions(e2e_app_api_client, app_permissions)
+
+    price = 10
+
+    (
+        warehouse_id,
+        channel_id,
+        channel_slug,
+        shipping_method_id,
+    ) = prepare_shop(e2e_staff_api_client)
+
+    channel_update_input = {
+        "orderSettings": {
+            "allowUnpaidOrders": True,
+            "automaticallyFulfillNonShippableGiftCard": True,
+            "automaticallyConfirmAllNewOrders": True,
+            "expireOrdersAfter": 1,
+            "deleteExpiredOrdersAfter": "1",
+        }
+    }
+    update_channel(
+        e2e_staff_api_client,
+        channel_id,
+        channel_update_input,
+    )
+
+    (
+        _product_id,
+        product_variant_id,
+        _product_variant_price,
+    ) = prepare_product(
+        e2e_staff_api_client,
+        warehouse_id,
+        channel_id,
+        price,
+    )
+
+    # Step 1 - Create checkout
+    lines = [
+        {
+            "variantId": product_variant_id,
+            "quantity": 1,
+        },
+    ]
+    checkout_data = checkout_create(
+        e2e_staff_api_client,
+        lines,
+        channel_slug,
+        email="testEmail@example.com",
+        set_default_billing_address=True,
+        set_default_shipping_address=True,
+    )
+    checkout_id = checkout_data["id"]
+    assert checkout_id is not None
+    assert checkout_data["isShippingRequired"] is True
+    assert len(checkout_data["shippingMethods"]) == 1
+    shipping_method_id = checkout_data["shippingMethods"][0]["id"]
+
+    # Step 2 - Assign shipping method
+    checkout_data = checkout_delivery_method_update(
+        e2e_staff_api_client,
+        checkout_id,
+        shipping_method_id,
+    )
+    assert checkout_data["deliveryMethod"]["id"] is not None
+
+    # Step 3 - Create order from the checkout
+    order_data = order_create_from_checkout(e2e_app_api_client, checkout_id)
+    order_id = order_data["order"]["id"]
+    assert order_id is not None
+    assert order_data["order"]["status"] == "UNCONFIRMED"
+    assert order_data["order"]["paymentStatus"] == "NOT_CHARGED"
+
+    # Step 4 - Wait and check the order is expired
+    time.sleep(80)
+    expire_orders_task()
+    data = order_query(e2e_staff_api_client, order_id)
+    assert data["status"] == "EXPIRED"
+    assert data["paymentStatus"] == "NOT_CHARGED"
+
+    # Step 5 - Wait and check the order has been deleted
+    time.sleep(80)
+    delete_expired_orders_task()
+    data = order_query(e2e_staff_api_client, order_id)
+    assert data["status"] == "EXPIRED"
+    assert data["paymentStatus"] == "NOT_CHARGED"
```

Comment:
I have some doubts here.
Shouldn't we expect that the order query does not return any data? 
time.sleep(80) means that we wait 80 seconds but deleteExpiredOrdersAfter is expressed in days so we should use something different here to make sure that the expiration date passed and check if the order does not exist.

Reply:
You're right - working on it now